modules/afsmtp/afsmtp-grammar.ym

```diff
@@ -75,25 +75,25 @@ afsmtp_options
 afsmtp_option
         : KW_HOST '(' string ')'		{ afsmtp_dd_set_host(last_driver, $3); free($3); }
         | KW_PORT '(' LL_NUMBER ')'		{ afsmtp_dd_set_port(last_driver, $3); }
-	| KW_SUBJECT '(' string ')'	 	{ afsmtp_dd_set_subject(last_driver, $3); free($3); }
-	| KW_BODY '(' string ')'		{ afsmtp_dd_set_body(last_driver, $3); free($3); }
+	| KW_SUBJECT '(' string ')'	 	{ afsmtp_dd_set_subject(last_driver, $3, configuration); free($3); }
+	| KW_BODY '(' string ')'		{ afsmtp_dd_set_body(last_driver, $3, configuration); free($3); }
 	| KW_HEADER '(' string string ')'	{
-		afsmtp_dd_add_header(last_driver, $3, $4);
+		afsmtp_dd_add_header(last_driver, $3, $4, configuration);
 		free($3); free($4);
```

Comment:
Please pass the configuration object to the constructor and don't specify it for all of these calls.